File: workshop3A9/src/main/java/Entites/Reclamation.java
package Entites;

import java.sql.Date;

public class Reclamation {
    private int id_reclamation;
    private int id_user;
    private String type_reclamation;
    private String sujet;
    private String description;
    private Date date_reclamation;
    private String statut;
    private String priorite;
    private String reponse_admin;
    private Date date_reponse;
    private String piece_jointe;

    public Reclamation() {
    }

    public Reclamation(int id_user, String type_reclamation, String sujet, String description, String priorite,
            String piece_jointe) {
        this.id_user = id_user;
        this.type_reclamation = type_reclamation;
        this.sujet = sujet;
        this.description = description;
        this.priorite = priorite;
        this.piece_jointe = piece_jointe;
        this.statut = "En attente";
        this.date_reclamation = new Date(System.currentTimeMillis());
    }

    public int getId_reclamation() {
        return id_reclamation;
    }

    public void setId_reclamation(int id_reclamation) {
        this.id_reclamation = id_reclamation;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }

    public String getType_reclamation() {
        return type_reclamation;
    }

    public void setType_reclamation(String type_reclamation) {
        this.type_reclamation = type_reclamation;
    }

    public String getSujet() {
        return sujet;
    }

    public void setSujet(String sujet) {
        this.sujet = sujet;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate_reclamation() {
        return date_reclamation;
    }

    public void setDate_reclamation(Date date_reclamation) {
        this.date_reclamation = date_reclamation;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    public String getPriorite() {
        return priorite;
    }

    public void setPriorite(String priorite) {
        this.priorite = priorite;
    }

    public String getReponse_admin() {
        return reponse_admin;
    }

    public void setReponse_admin(String reponse_admin) {
        this.reponse_admin = reponse_admin;
    }

    public Date getDate_reponse() {
        return date_reponse;
    }

    public void setDate_reponse(Date date_reponse) {
        this.date_reponse = date_reponse;
    }

    public String getPiece_jointe() {
        return piece_jointe;
    }

    public void setPiece_jointe(String piece_jointe) {
        this.piece_jointe = piece_jointe;
    }

    @Override
    public String toString() {
        return "Reclamation{" +
                "id=" + id_reclamation +
                ", sujet='" + sujet + '\'' +
                ", statut='" + statut + '\'' +
                '}';
    }
}

File: src/Entity/Reclamation.php
<?php

namespace App\Entity;

use App\Repository\ReclamationRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ReclamationRepository::class)]
#[ORM\Table(name: 'reclamation')]
class Reclamation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_reclamation', type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(name: 'id_user', type: 'integer')]
    private int $idUser = 0;

    #[ORM\Column(name: 'type_reclamation', type: 'string', length: 50, nullable: true)]
    private ?string $typeReclamation = null;

    #[ORM\Column(type: 'string', length: 100, nullable: true)]
    private ?string $sujet = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(name: 'date_reclamation', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateReclamation = null;

    #[ORM\Column(type: 'string', length: 30, options: ['default' => 'En attente'])]
    private string $statut = 'En attente';

    #[ORM\Column(type: 'string', length: 20, nullable: true)]
    private ?string $priorite = null;

    #[ORM\Column(name: 'piece_jointe', type: 'string', length: 255, nullable: true)]
    private ?string $pieceJointe = null;

    #[ORM\Column(name: 'reponse_admin', type: 'text', nullable: true)]
    private ?string $reponseAdmin = null;

    #[ORM\Column(name: 'date_reponse', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateReponse = null;

    public function __construct() { $this->dateReclamation = new \DateTime(); }

    public function getId(): ?int { return $this->id; }
    public function getIdUser(): int { return $this->idUser; }
    public function setIdUser(int $idUser): static { $this->idUser = $idUser; return $this; }
    public function getTypeReclamation(): ?string { return $this->typeReclamation; }
    public function setTypeReclamation(?string $typeReclamation): static { $this->typeReclamation = $typeReclamation; return $this; }
    public function getSujet(): ?string { return $this->sujet; }
    public function setSujet(?string $sujet): static { $this->sujet = $sujet; return $this; }
    public function getDescription(): ?string { return $this->description; }
    public function setDescription(?string $description): static { $this->description = $description; return $this; }
    public function getDateReclamation(): ?\DateTimeInterface { return $this->dateReclamation; }
    public function setDateReclamation(?\DateTimeInterface $dateReclamation): static { $this->dateReclamation = $dateReclamation; return $this; }
    public function getStatut(): string { return $this->statut; }
    public function setStatut(string $statut): static { $this->statut = $statut; return $this; }
    public function getPriorite(): ?string { return $this->priorite; }
    public function setPriorite(?string $priorite): static { $this->priorite = $priorite; return $this; }
    public function getPieceJointe(): ?string { return $this->pieceJointe; }
    public function setPieceJointe(?string $pieceJointe): static { $this->pieceJointe = $pieceJointe; return $this; }
    public function getReponseAdmin(): ?string { return $this->reponseAdmin; }
    public function setReponseAdmin(?string $reponseAdmin): static { $this->reponseAdmin = $reponseAdmin; return $this; }
    public function getDateReponse(): ?\DateTimeInterface { return $this->dateReponse; }
    public function setDateReponse(?\DateTimeInterface $dateReponse): static { $this->dateReponse = $dateReponse; return $this; }
}
